deployment/ccip/deploy.go

```diff
@@ -262,10 +265,28 @@ func DeployPrerequisiteContracts(e deployment.Environment, ab deployment.Address
 			return err
 		}
 		e.Logger.Infow("deployed router", "addr", routerContract.Address)
-		r = routerContract.Contract
 	} else {
 		e.Logger.Infow("router already deployed", "addr", chainState.Router.Address)
 	}
+	if mc3 == nil {
```

Comment:
Do we need to deploy mc3 always? Or should it have some conditional input?

Reply:
What's the best way to condition it?

Reply:
We use `DeployCCIPContractConfig` to tell the deployer whether to deploy or not USDC-related contracts.
https://github.com/smartcontractkit/chainlink/blob/43f0ff1334b2e0c12b5073125dc524874839b70f/deployment/ccip/deploy.go#L339

Not sure if it fits your case. Another way would be to deploy multicall explicitly on demand (using a separate function)

```go
ccdeploy.DeployMultiCall(chainId)
```